### PR TITLE
Make coordinate optional for AddressAutofill/PlaceAutocomplete

### PR DESCRIPTION
**Pull request by eugenes78**
### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR (where applicable)
- [x] I have run tests and automatic checks locally
- [x] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [x] I have made corresponding changes to the documentation (where applicable)
- [x] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

File: MapboxSearch/autofill/src/test/java/com/mapbox/search/autofill/TypeObjectCreator.kt [hunk 1]
```diff
@@ -28,7 +29,13 @@ internal object TypeObjectCreator {
         )
 
         listOf(
-            AddressAutofillSuggestion("name", "formattedAddress", addressComponents, null)
+            AddressAutofillSuggestion(
+                name = "name",
+                formattedAddress = "formattedAddress",
+                coordinate = Point.fromLngLat(10.0, 15.0),
+                address = addressComponents,
+                underlying = null
```
Comment (mapbox-danny) [inline MapboxSearch/autofill/src/test/java/com/mapbox/search/autofill/TypeObjectCreator.kt:37]: Is this line added intentionally?

Comment (eugenes78) [reply MapboxSearch/autofill/src/test/java/com/mapbox/search/autofill/TypeObjectCreator.kt:37]: Yes, this is a test (note `.../src/test/...` in the path)

File: MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteSuggestion.kt [hunk 2]
```diff
@@ -88,6 +95,7 @@ public class PlaceAutocompleteSuggestion internal constructor(
     override fun hashCode(): Int {
         var result = name.hashCode()
         result = 31 * result + (formattedAddress?.hashCode() ?: 0)
+        result = 31 * result + coordinate.hashCode()
```
Comment (mikeringrose) [inline MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteSuggestion.kt:98]: can't `coordinate` be null here?

**Review by mikeringrose — COMMENTED** [review]
just have one question.

File: MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteSuggestion.kt (hunk 2 above)
Comment (eugenes78) [reply MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteSuggestion.kt:98]: Good catch! It'll be null in most cases. Let me fix that real quick.

Comment (eugenes78) [reply MapboxSearch/place-autocomplete/src/main/java/com/mapbox/search/autocomplete/PlaceAutocompleteSuggestion.kt]: Fixed the same in both AddressAutofill and PlaceAutocomplete

**Review by mikeringrose — APPROVED** [review]

**Pull request merged**